### PR TITLE
Deep CNTK Integration

### PR DESCRIPTION
**Pull request by chdinh**

### CODE REVIEW

File: MNE/disp/deepmodelviewer/network.cpp [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+}
+
+
+//*************************************************************************************************************
+
+void Network::updateWeights()
+{
+    // Update Weights
+
+    if(!m_pModel)
```
Comment (LorenzE) [inline MNE/disp/deepmodelviewer/network.cpp]: please always use brackets

File: MNE/disp/deepmodelviewer/network.cpp [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+        }
+    }
+    layerDim.append(outDim);
+
+    //
+    // Create items according to the dimensions
+    //
+
+    for(int layer = 0; layer < layerDim.size(); ++layer) {
+
```
Comment (LorenzE) [inline MNE/disp/deepmodelviewer/network.cpp]: remove empty lines

File: MNE/disp/deepmodelviewer/network.cpp [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+    }
+}
+
+
+//*************************************************************************************************************
+
+void Network::generateNetwork()
+{
+
+    if(!m_pModel)
```
Comment (LorenzE) [inline MNE/disp/deepmodelviewer/network.cpp]: brackets

File: MNE/disp/deepmodelviewer/network.h [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    */
+    bool isSetup() const;
+
+    //=========================================================================================================
+    /**
+    * Returns the current setup pen style
+    *
+    * @return the current pen style
+    */
+    inline Qt::PenStyle getPenStyle() const;
```
Comment (LorenzE) [inline MNE/disp/deepmodelviewer/network.h]: mind empty lines as spacers in the couple next lines

File: MNE/disp/deepmodelviewer/network.h [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+    */
+    inline float weightStrength() const;
+
+    //=========================================================================================================
+    /**
+    * Update the weights according to the current model
+    */
+    void updateWeights();
+
+signals:
```
Comment (LorenzE) [inline MNE/disp/deepmodelviewer/network.h]: signals go to the bottom after private

File: applications/mne_analyze/libs/anShared/Management/analyzedata.h [hunk 6]
```diff
@@ -100,10 +109,39 @@ class ANSHAREDSHARED_EXPORT AnalyzeData : public QObject
     */
     virtual ~AnalyzeData();
 
-protected:
+// Database Interface for now -> consider using a Abstract Item Model
+signals:
```
Comment (LorenzE) [inline applications/mne_analyze/libs/anShared/Management/analyzedata.h]: signals go to the bottom

File: MNE/deep/deep.h [hunk 7]
```diff
@@ -236,12 +246,17 @@ class DEEPSHARED_EXPORT Deep
...
 
+    void cancelTraining();
+
     //=========================================================================================================
     /**
     * Print the model structure
     */
     void print();
 
+signals:
```
Comment (chdinh) [inline MNE/deep/deep.h]: not okay I think

Comment (chdinh) [inline MNE/deep/deep.h]: And what also should be considered to have multiple public private sections, which are grouping certain functionalities together - at the end of each such group there should reside the respective signals

File: MNE/disp/deepmodelviewer/network.h (hunk 5 above)
Comment (chdinh) [inline MNE/disp/deepmodelviewer/network.h]: do not agree

File: applications/mne_analyze/libs/anShared/Management/analyzedata.h (hunk 6 above)
Comment (chdinh) [inline applications/mne_analyze/libs/anShared/Management/analyzedata.h]: not agree

**Pull request merged**